### PR TITLE
python311Packages.hdfs: 2.5.8 -> 2.6.0

### PR DESCRIPTION
**Pull request by fabaff**
Diff: https://github.com/mtth/hdfs/compare/refs/tags/2.5.8...2.6.0

Changelog: https://github.com/mtth/hdfs/releases/tag/2.6.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/hdfs/default.nix [hunk 1]
```diff
@@ -1,35 +1,63 @@
-{ buildPythonPackage
+{ lib
+, buildPythonPackage
```
Comment (samuela) [inline pkgs/development/python-modules/hdfs/default.nix:2]: let's please keep these arguments alphabetical

File: pkgs/development/python-modules/hdfs/default.nix [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
+    rev = "refs/tags/${version}";
+    hash = "sha256-r3zt0upD+wzHZh8ktFpztcz3agM5sjz+nb/19ErD3ko=";
   };
 
-  propagatedBuildInputs = [ docopt requests six ];
+  propagatedBuildInputs = [
+    docopt
+    requests
+    six
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/hdfs/default.nix:33]: let's avoid polluting the git blame

File: pkgs/development/python-modules/hdfs/default.nix [hunk 3]
```diff
@@ -1,35 +1,63 @@
...
-  nativeCheckInputs = [ fastavro nose pytestCheckHook ];
+  nativeCheckInputs = [
+    nose
+    pytestCheckHook
+  ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
 
-  pythonImportsCheck = [ "hdfs" ];
+  pythonImportsCheck = [
+    "hdfs"
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/hdfs/default.nix:55]: let's avoid polluting the git blame

**Review by samuela — CHANGES_REQUESTED** [review]

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 235074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.apache-beam</li>
    <li>python311Packages.apache-beam.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.apache-beam</li>
    <li>python310Packages.apache-beam.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.hdfs</li>
    <li>python310Packages.hdfs.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.hdfs</li>
    <li>python311Packages.hdfs.dist</li>
  </ul>
</details>

**Pull request closed without merging**